test/built-ins/AsyncGeneratorPrototype/next/request-queue-await-order.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2018 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-asyncgenerator-prototype-next
+description: next() requests from iterator processed in order, await
+info: >
+  AsyncGenerator.prototype.next ( value )
+  1. Let generator be the this value.
+  2. Let completion be NormalCompletion(value).
+  3. Return ! AsyncGeneratorEnqueue(generator, completion).
+
+  AsyncGeneratorEnqueue ( generator, completion )
+  ...
+  4. Let queue be generator.[[AsyncGeneratorQueue]].
+  5. Let request be AsyncGeneratorRequest{[[Completion]]: completion,
+     [[Capability]]: promiseCapability}.
+  6. Append request to the end of queue.
+  ...
+
+  AsyncGeneratorResolve ( generator, value, done )
+  ...
+  2. Let queue be generator.[[AsyncGeneratorQueue]].
+  3. Assert: queue is not an empty List.
+  4. Remove the first element from queue and let next be the value of that element.
+  ...
+
+flags: [async]
+features: [async-iteration]
+---*/
+
+var yieldorder = 0;
+
+function resolveInTime(t) {
+  return new Promise(resolve => {
+    setTimeout(() => {
```

Comment:
setTimeout is not part of Ecma262, so we cannot rely on it for the test, despite the much we would like to.

Reply:
oh wow -- hmm this test still worth doing without the waiting, or is there another way to do it? Maybe we can talk Monday.

Reply:
In an earlier meeting today, @spectranaut decided to try replacing the setTimeout cases with a promise to be resolved after the function scope.

Maybe you can expose resolve to a outer scope name and resolve it after the resolveInTime function...